### PR TITLE
Add ArrayExpression support in related tables

### PR DESCRIPTION
**Pull request by np25071984**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | Fixed error which obtained during accessing to related data Array type. You can find user case in #16966

### CODE REVIEW

File: framework/db/ActiveRelationTrait.php [hunk 1]
```diff
@@ -525,6 +525,8 @@ private function filterByModels($models)
                 if (($value = $model[$attribute]) !== null) {
                     if (is_array($value)) {
                         $values = array_merge($values, $value);
+                    } elseif(($value instanceof ArrayExpression) && ($value->getDimension() === 1)) {
```
Comment (rob006) [inline framework/db/ActiveRelationTrait.php]:
```suggestion
                    } elseif ($value instanceof ArrayExpression && $value->getDimension() === 1) {
```

Comment (GHopperMSK) [reply framework/db/ActiveRelationTrait.php]: No, unfortunately it doesn't help

**Comment by rob006** [comment]
I would ignore this Codeclimate complaining - there is no point to create code less readable code for humans only to satisfy SCA tool.

**Comment by GHopperMSK** [comment]
> I would ignore this Codeclimate complaining - there is no point to create code less readable code for humans only to satisfy SCA tool.

In this case I think I should undo last commit due to it isn't such readable for human as previous one.

**Review by SilverFire — APPROVED** [review]
Thank you for a pull request. It looks perfectly good to me

**Pull request merged**